src/App.js

```diff
@@ -9,94 +9,97 @@ class App extends Component {
+    const isCarousel = event.target.closest('.Carousel');
+    const popUpGenre = isCarousel && event.target.closest('.Carousel').dataset.genre;
+
     const popUpInfo = this.state.games
       .find(game => {
         return game.img === event.target.src;
-      })
-    this.setState({ popUpInfo, popUpGenreID: genreID });
+      });
+
+    this.setState({ popUpInfo, popUpGenre });
   }
 
   closePopUp = () => {
-    this.setState({ popUpInfo: null })
+    this.setState({
+      popUpInfo: null,
+      popUpGenre: null,
+    })
   }
 
   checkFilterInput = (event) => {
     let inputValue = event.target.value.toLowerCase();
-    this.setState({searching: inputValue});
-    let filteredGames = this.state.games
+    const filteredGames = this.state.games
       .filter(game => game.game.toLowerCase().includes(inputValue))
-      .sort((a,b) => {
-        if (a.game > b.game){return 1} 
-        if (a.game < b.game) {return -1} 
-        return 0;})
-    this.setState({filteredGames: filteredGames});
+      .sort((a, b) => a.game.localeCompare(b.game));
+
+    this.closePopUp();
+    this.setState({
+      searching: inputValue,
+      filteredGames
+    });
   }
 
   render() {
-    let { error, areGamesLoaded, areGenresLoaded, games, 
-          genres, popUpInfo, popUpGenreID, searching, filteredGames } = this.state;
-    if (error) {
-      return <div>SOMETHING IS BAD 💩 </div>
-    } else if (areGamesLoaded && areGenresLoaded) {
+    let { errors, games,
+      genres, popUpInfo, popUpGenre, searching, filteredGames } = this.state;
+
+    if (genres && games && !errors) {
       return (
         <div className="App">
-          <Navbar checkFilterInput={this.checkFilterInput}/>
-          {searching ? 
-            <SearchPage 
-              filteredGames={filteredGames}
-              popUpInfo={popUpInfo}
-              createPopUp={this.createPopUp}
-              closePopUp={this.closePopUp} /> :
-            <LandingPage
-              genres={genres}
-              games={games}
-              popUpInfo={popUpInfo}
-              popUpGenreID={popUpGenreID}
-              createPopUp={this.createPopUp}
-              closePopUp={this.closePopUp} />
+          <Navbar checkFilterInput={this.checkFilterInput} />
+          {
+            searching ?
+              <SearchPage
+                filteredGames={filteredGames}
+                popUpInfo={popUpInfo}
+                createPopUp={this.createPopUp}
+                closePopUp={this.closePopUp} />
+              :
+              <LandingPage
+                genres={genres}
+                games={games}
+                popUpInfo={popUpInfo}
+                popUpGenre={popUpGenre}
+                createPopUp={this.createPopUp}
+                closePopUp={this.closePopUp} />
           }
         </div>
       );
+    }
+
+    if (errors) {
+      return <span role={"img"} aria-label="error">SOMETHING IS BAD 💩 </span>
     } else {
       return (<div>LOADING</div>)
     }
```

Comment:
This looks great, it makes more sense to have it in this order. Thanks for taking the time to adjust this.